scripts/ci/release_supply_chain.py

```
@@ -0,0 +1,568 @@
+}
+FORBIDDEN_KEY_RE = re.compile(r"(?:token|secret|password|private.?key)", re.IGNORECASE)
+FORBIDDEN_VALUE_RE = re.compile(
+    r"(?:github_pat_|gh[oprsu]_[A-Za-z0-9_]{20,}|-----BEGIN [A-Z ]*PRIVATE KEY-----)"
+)
+
+
+class ReleaseEvidenceError(ValueError):
+    """Raised when release supply-chain evidence is inconsistent."""
+
+
+def fail(message: str) -> "NoReturn":
```

Ruff (0.16.3)

[error] Undefined name `NoReturn`

(F821)